pkg/didcomm/common/service/did_comm_msg.go

```diff
@@ -23,8 +23,22 @@ const (
 	jsonType     = "@type"
 	jsonThread   = "~thread"
 	jsonThreadID = "thid"
+	jsonMetadata = "metadata"
```

Comment:
This should be a prefixed field to avoid collision with a real field named metadata.

Reply:
I've changed `metadata` to `internal_metadata`